crawler/src/main/java/com/movie/recommender/crawler/service/MovieService.java

```diff
@@ -1,25 +1,192 @@
+            }
+        }
+
+        // Process movie details for all now playing movies
+        processNowPlayingMoviesDetails(nowPlayingMoviesList);
+    }
+
+
+    public void processNowPlayingMoviesDetails(List<NowPlayingMoviesByCountry> nowPlayingMoviesList) {
+        List<MovieDetails> movieDetailsList = new ArrayList<>();
+
+        for (NowPlayingMoviesByCountry nowPlayingMovies : nowPlayingMoviesList) {
+            for (NowPlayingMoviesByCountry.MovieIdentifier movie : nowPlayingMovies.getResults()) {
+                // Fetch movie details using the new method
+                Optional<MovieDetails> movieDetailsOpt = fetchMovieDetails(movie.getId());
+                movieDetailsOpt.ifPresent(movieDetailsList::add);
+            }
+        }
+
+        // Insert all collected movie details into MongoDB
+        mongoDBService.insertMovies(movieDetailsList);
+        log.info("All movie details inserted into the 'movies' collection.");
     }
 
-    // For testing purposes. Verify that the country with cities is retrieved.
-    // Should be removed later.
-    public void getCountryWithCities() {
-        log.info("Getting country with cities...");
-        locationServiceClient.getAllCountriesAndCities().forEach(countryWithCitiesDTO -> {
-            log.info("Country: {}", countryWithCitiesDTO.getCountryName());
-            countryWithCitiesDTO.getCities().forEach(city -> log.info("City: {}", city));
+    private Optional<MovieDetails> fetchMovieDetails(Long movieId) {
+        // Check if the movie already exists in MongoDB
+        if (mongoDBService.existsMovieById(movieId)) {
+            log.info("Movie ID: {} already exists in MongoDB. Skipping...", movieId);
+            return Optional.empty();
+        }
+
+        log.info("Processing Movie ID: {}", movieId);
+
+        // Load movie details
+        Optional<MovieDetails> movieDetailsOpt = tmdbApiClient.getMovieDetails(movieId.toString());
+        if (movieDetailsOpt.isEmpty()) {
+            log.warn("No details found for movie ID: {}", movieId);
+            return Optional.empty();
+        }
+
+        MovieDetails movieDetails = movieDetailsOpt.get();
+
+        // Load movie credits
+        Optional<MovieCredits> creditsOpt = tmdbApiClient.getMovieCredits(movieId.toString());
+        creditsOpt.ifPresent(credits -> {
+            movieDetails.setCast(credits.getCast());
+            movieDetails.setCrew(credits.getCrew());
         });
+
+        // Load movie keywords
+        Optional<MovieKeywords> keywordsOpt = tmdbApiClient.getMovieKeywords(movieId.toString());
+        keywordsOpt.ifPresent(keywords -> movieDetails.setKeywords(keywords.getKeywords()));
+
+        return Optional.of(movieDetails);
+    }
+
+
+    public void loadShowtimes() {
+        Map<String, String> countryCodeMapping = Map.of(
+                "Ukraine", "ua"
+        );
+
+        List<NowPlayingMoviesByCountry> nowPlayingMovies = mongoDBService.getNowPlayingMovies();
```

Comment:
you already have now playing movies
instead of loading them again from mongo, return the list from `loadNowPlayingMovies` and pass it as a parameter to `loadShowtimes`